internal/trace/transport.go

```diff
@@ -16,56 +16,58 @@ limitations under the License.
 package trace
 
 import (
+	"fmt"
 	"net/http"
 	"strings"
-
-	"github.com/sirupsen/logrus"
+	"sync/atomic"
 )
 
 // Transport is an http.RoundTripper that keeps track of the in-flight
 // request and add hooks to report HTTP tracing events.
 type Transport struct {
 	http.RoundTripper
+	count uint64
 }
 
 func NewTransport(base http.RoundTripper) *Transport {
-	return &Transport{base}
+	return &Transport{base, 0}
 }
 
 // RoundTrip calls base roundtrip while keeping track of the current request.
 func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	number := atomic.AddUint64(&t.count, 1)
 	ctx := req.Context()
 	e := Logger(ctx)
 
-	e.Debugf("> Request URL: %q", req.URL)
-	e.Debugf("> Request method: %q", req.Method)
-	e.Debugf("> Request headers:")
-	logHeader(req.Header, e)
+	// log the request
+	e.Debugf("\nRequest #%d\n> Request URL: %q\n> Request method: %q\n> Request headers:\n%s",
```

Comment:
nit: the initial line return can be removed
```suggestion
	e.Debugf("Request #%d\n> Request URL: %q\n> Request method: %q\n> Request headers:\n%s",
```

Reply:
removed